play/src/front/Components/Video/PictureInPicture.svelte

```diff
@@ -0,0 +1,259 @@
+                                    videoElement.style.zIndex = "100";
+                                    videoElement.style.position = "absolute";
+                                    videoElement.style.top = "0";
+                                    videoElement.style.left = "0";
+                                    videoElement.style.right = "0";
+                                    videoElement.style.bottom = "0";
+                                    videoElement.style.backgroundColor = "#373a3e";
+                                    videoElement.style.aspectRatio = `${ratio}`;
+
+                                    if (divMailElement) divMailElement.appendChild(videoElement);
+                                }, 1000);
+                            })
+                        );
+                    } catch (e) {
+                        console.info(
+                            "Source undefined. The element streamable will not attach to the video element",
+                            e,
+                            streamable.uniqueId
+                        );
+                    }
+                }
+            });
+
+            // start the subscrbtion to active my camera
+            unsubscribeLocalStreamStore = localStreamStore.subscribe((value) => {
+                if (value.type === "success") {
+                    if (value.stream == undefined) return;
+                    myLocalStream = value.stream;
+                    // TODO: remove this hack
+                    setTimeout(() => {
+                        if (myLocalStreamElement)
+                            myLocalStreamElement.style.aspectRatio = `${
+                                value.stream?.getVideoTracks()[0]?.getSettings().aspectRatio
+                            }`;
+                    }, 100);
+                } else {
+                    myLocalStream = null;
+                }
+            });
+        });
+    }
+
+    function destroyPictureInPictureComponent() {
+        if ($pipWindowStore) $pipWindowStore.removeEventListener("pagehide", destroyPictureInPictureComponent);
+        if ($pipWindowStore) $pipWindowStore.close();
+        pipWindowStore.set(undefined);
+        if (streamableCollectionStoreSubscriber) streamableCollectionStoreSubscriber();
+        if (activePictureInPictureSubscriber) activePictureInPictureSubscriber();
+        if (unsubscribeLocalStreamStore) unsubscribeLocalStreamStore();
+        streamablesCollectionStoreSubscriber.forEach((subscriber) => subscriber());
+        activePictureInPictureStore.set(false);
+    }
+
+    function getPlayerWokaPicture(streamable: Streamable): string {
+        const gameScene = gameManager.getCurrentGameScene();
+        const playerWokaPictureStore = [...gameScene.MapPlayersByKey].find(
+            ([, player]) => player.userUuid === (streamable as VideoPeer).player.userUuid
+        )?.[1].pictureStore;
```

Comment:
The streamable contains a method to directly fetch the picture.

You can simply do: `streamable.pictureStore`

https://github.com/thecodingmachine/workadventure/blob/c851612079c358876cac5c8c077dd16d0c30301e/play/src/front/Components/Video/VideoMediaBox.svelte#L32-L32

Reply:
Hum... in the new version ?
Because on master or develop doesn't exist yet 😭